### PR TITLE
Add simple cli key generator

### LINKED ISSUE #219: Feature: Add to CLI a easy key generator for RSTUF Users
### What do you want to share with us?

A user raised an [interesting question in the #repository-service-for-tuf channel](https://cloud-native.slack.com/archives/C047L55314N/p1678137630038059). What we notice is a valuable user experience.

When we search about keys generation, most of the instructions use `ssh-keygen` or `openssl`.

RSTUF uses [securesystemslib](https://github.com/secure-systems-lab/securesystemslib) to manage the Keys in the CLI and the Worker. However, currently, securesystemslib doesn't support those key files.

This issue's goal is:
* Add to the CLI a feature to generate keys using [securesystemlib](https://github.com/secure-systems-lab/securesystemslib) and make it easy to use.
* Ask the user for the file name
* Ask the user about the algorithm
* Ask the user for the password

```command
$ rstuf key generate
```

### References

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KAUTH**
This newly added command is used to generate cryptographic keys using the `securesystemslib` library.

Notes:
- We probably still need to create a documentation task for this new CLI command

Closes #219

### CODE REVIEW

File: repository_service_tuf/cli/generate_key.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        f"\nChoose key type",
+        choices=KeyType.get_all_members(),
+        default=KeyType.KEY_TYPE_ED25519.value,
+    )
+
+    filepath: str = prompt.Prompt.ask(
+        "Enter the keys' [green]filename[/] (default path is the current "
+        "working directory) to store the files",
+        default="./rstuf-key",
+    )
```
Comment (kairoaraujo) [inline repository_service_tuf/cli/generate_key.py:34]: In the implementation, the default path is not the current working directory, it actually defines a filename "rstuf-key".
I suggest only asking the user for the filename and giving the default filename as  `id_{key_type}` following the same standard as `ssh-keygen`. ;)

Additionally, I would suggest an improvement in the user experience, checking if the key already exists and asking if want to overwrite (same as `ssh-keygen`).


```suggestion
    filepath: str = prompt.Prompt.ask(
        "Enter the key's [green]filename[/]:",
        default=f"id_{key_type}",
    )
    if os.path.isfile(filepath) is True:
        overwrite = prompt.Confirm.ask(
            f"Do you want [red]overwrite[/] the existent '{filepath}' file?"
        )
        if overwrite is False:
            raise click.ClickException("Key creation aborted.")    
```

**Review by kairoaraujo — CHANGES_REQUESTED** [review]
I loved it! ❤️ 

I'm anticipating my review because. It will help improve my documentation https://github.com/vmware/repository-service-tuf/pull/284

**Comment by kairoaraujo** [comment]
The suggestions I made above will look like this:

```shell
❯ rstuf generate-key

Choose key type [ed25519/ecdsa/rsa] (ed25519): rsa
Enter the key's filename: (id_rsa): 
Do you want overwrite the existent 'id_rsa' file? [y/n]: y
enter password to encrypt private key file 'id_rsa' (leave empty if key should not be encrypted): 
```

**Comment by kairoaraujo** [comment]
Could you please also update the documentation with the `generate-key`?
`docs/source/guide/index.rst`

File: repository_service_tuf/cli/generate_key.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            generate_and_write_ed25519_keypair_with_prompt(filepath)
+
+        case KeyType.KEY_TYPE_ECDSA.value:
+            generate_and_write_ecdsa_keypair_with_prompt(filepath)
+
+        case KeyType.KEY_TYPE_RSA.value:
+            generate_and_write_rsa_keypair_with_prompt(filepath)
+
+        case _:
+            raise ValueError(f"Key type `{key_type}` is not supported!")
```
[the remark this answers is not in the document]
Comment (KAUTH) [reply repository_service_tuf/cli/generate_key.py:47]: Indeed, I added just for the sake of completeness and perhaps catching any stray bug in the future :stuck_out_tongue: .

File: repository_service_tuf/cli/generate_key.py (hunk 1 above)
Comment (KAUTH) [reply repository_service_tuf/cli/generate_key.py:34]: I like the suggestion! It is much clearer now as well

Comment (kairoaraujo) [reply repository_service_tuf/cli/generate_key.py:34]: My option would be `id` as I suggested above, just to be standard. As keys are linked to identification (people, roles, etc.).
But it is just my taste :)

File: repository_service_tuf/cli/generate_key.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    match key_type:
+        case KeyType.KEY_TYPE_ED25519.value:
+            generate_and_write_ed25519_keypair_with_prompt(filepath)
+
+        case KeyType.KEY_TYPE_ECDSA.value:
+            generate_and_write_ecdsa_keypair_with_prompt(filepath)
+
+        case KeyType.KEY_TYPE_RSA.value:
+            generate_and_write_rsa_keypair_with_prompt(filepath)
```
Comment (kairoaraujo) [inline repository_service_tuf/cli/generate_key.py:44]: Hey @KAUTH,

We shouldn't use the `_with_prompt` from securesystemslib.
The reason is that we use encrypted keys always in the Worker/CLI Ceremony for now.
So, securesystemslib makes it optional, whereas in our use case is mandatory.

**Comment by kairoaraujo** [comment]
I rebased your PR on top of the main. I did it once we did merge `roles-simplification` branch to the `main`
It is ready to continue.

**Comment by KAUTH** [comment]
> I rebased your PR on top of the main. I did it once we did merge `roles-simplification` branch to the `main` It is ready to continue.

Great thanks, I pushed the new changes. I didn't add the changes to `docs/source/guide/index.rst` yet until we finalize the command and its output.

**Comment by kairoaraujo** [comment]
@KAUTH LGTM, adding the docs is ready to merge 🚀

**Review by kairoaraujo — CHANGES_REQUESTED** [review]
Include documentation to the guide `docs/source/guide/index.rst`

File: docs/source/guide/index.rst [hunk 4]
```diff
@@ -541,4 +543,57 @@ See the below CSV file example:
     Import status: Publish targets task id is dd1cbf2320ad4df6bda9ca62cdc0ef82
     Import status: task STARTED
     Import status: task SUCCESS
-    Import status: Finished.
\ No newline at end of file
+    Import status: Finished.
+
+
+Key Management (``key``)
+==========================
```
Comment (kairoaraujo) [inline docs/source/guide/index.rst:550]:
```suggestion
Key Management (``key``)
========================
```

File: docs/source/guide/index.rst [hunk 5]
```diff
@@ -541,4 +543,57 @@ See the below CSV file example:
...
+
+    ╭─ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
+    │ --help  -h    Show this message and exit.                                                                                        │
+    ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+    ╭─ Commands ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
+    │ generate                        Generate cryptographic keys using the `securesystemslib` library                                 │
+    ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+
+Key Generation (``generate``)
+---------------------------
```
Comment (kairoaraujo) [inline docs/source/guide/index.rst:573]:
```suggestion
Key Generation (``generate``)
-----------------------------
```

File: docs/source/guide/index.rst [hunk 6]
```diff
@@ -541,4 +543,57 @@ See the below CSV file example:
...
+3. a password, to encrypt the private key file
+
+After the above procedure, two files, the private and public key
+(e.g., ``id_ed25519`` and ``id_ed25519.pub``), will be generated in the current
+working directory.
+
+The generated keys may be used in the Repository Service for TUF Ceremony
+process, for the online key or the TUF roles' keys (``root``, ``targets``, etc. keys).
+
+.. code:: shell
```
Comment (kairoaraujo) [inline docs/source/guide/index.rst:593]: It breaks the build 

```shell
/Users/kdearaujo/devel/RSTUF/repository-service-tuf-cli/docs/source/guide/index.rst:593:Could not lex literal_block as "shell". Highlighting skipped.
```

```suggestion
.. code::
```

**Review by kairoaraujo — CHANGES_REQUESTED** [review]
I suggested some changes to ensure the make docs pass in the Umbrella.

Additionally, we need to add the new module to the development doc as well.

`docs/source/devel/index.rst` add at line 20:
`repository_service_tuf.cli.key`

It will look like
```rst
.. toctree::
   :maxdepth: 4

   repository_service_tuf
   repository_service_tuf.cli
   repository_service_tuf.cli.admin
   repository_service_tuf.cli.key
   repository_service_tuf.helpers
   modules
```

File: docs/source/guide/index.rst (hunk 6 above)
Comment (KAUTH) [reply docs/source/guide/index.rst]: Indeed, although there are other `.. code:: shell` in the code :thinking:. In any case, I'm changing it for now

**Review by kairoaraujo — APPROVED** [review]

**Comment by kairoaraujo** [comment]
Amazing work @KAUTH 🎉

**Pull request merged**